torch/utils/_cxx_pytree.py

```diff
@@ -26,8 +26,19 @@
     Union,
 )
 
-import optree
-from optree import PyTreeSpec  # direct import for type annotations
+import torch
+from ._pytree import TreeSpec as PyTreeSpec
+
+try:
+    if torch._running_with_deploy() and torch.compiled_with_cxx11_abi():
+        raise ModuleNotFoundError
```

Comment:
In the test CI for torch deploy, the torch libraries are built with `_GLIBCXX_USE_CXX11_ABI=1`.

Ref:

- #51039